Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -482,6 +520,35 @@ private void OnTransferDnaEvent(EntityUid uid, DnaComponent component, ref Trans
 
         #region Public API
 
+        #region Goobstation Fingerprint Methods
+
+        /// <summary>
+        /// Merges an existing partial fingerprint with a full fingerprint, revealing some characters randomly based on the reveal
+        /// chance.
+        /// </summary>
+        private string CreateOrMergePartialFingerprintRandomly(string full, string? value)
+        {
+            var valueX = value ?? new string('#', full.Length);
+            var merged = string.Empty;
+            var ix = 0;
+            var revealed = false;
+            foreach (var i in valueX)
+            {
+                if (i != '#') //if already revealed
+                    merged += i; // keep revealed
+                else if (_random.Prob(_revealChance)) // reveal based on chance
+                {
+                    merged += full[ix]; // reveal
+                    revealed = true;
+                }
+                else
+                    merged += '#'; // hide them all if they are not revealed
+                ix++;
+            }
+            return revealed ? new string(merged) : string.Empty; // return empty value if  nothing was revealed
+        }
```

Comment:
dude what the fuck is this shit why

```suggestion
	private string CreateOrMergePartialFingerprintRandomly(string full, string? currentMask)
	{
	    if (string.IsNullOrEmpty(full)) 
	    	return string.Empty;

	    char[] buffer = currentMask?.ToCharArray() ?? new string('#', full.Length).ToCharArray();
	
	    if (buffer.Length != full.Length) 
	    	return string.Empty;
	
	    var revealedAnyNew = false;
	
	    for (var i = 0; i < buffer.Length; i++)
	    {
	        if (buffer[i] == '#')
	        {
	            if (_random.Prob(_revealChance))
	            {
	                buffer[i] = full[i];
	                revealedAnyNew = true;
	            }
	        }
	    }

	    return revealedAnyNew ? new string(buffer) : string.Empty;
	}
```